### ISSUE TITLE
Image block: linked image do not have a pressed button in the block toolbar

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#56078]
### Description

When some normal text in the editor is linked, the 'Link' button in the block toolbar communicates there is a link both visually and semantically. Screenshot:

![Screenshot 2023-11-13 at 10 30 20](https://github.com/WordPress/gutenberg/assets/1682452/59790627-0d50-4aff-95bb-1b244b9e71fd)

Instead, when an image is linked, the 'Link' button doesn't communicate the 'linked' stage. Screenshot:

![Screenshot 2023-11-13 at 10 30 43](https://github.com/WordPress/gutenberg/assets/1682452/3c36b186-6549-4fb0-b487-460300ec7b09)

the button misses the 'is-pressed' CSS class and the `aria-pressed="true"` attribute.

Besides that, labelling, functionality and UI to unlink/edit link are completely different for text and images. Ideally, the UIs should be similar but that's not the scope of this issue.

### Step-by-step reproduction instructions

- Edit a post.
- Select some text and add a link. Save.
- Select the selected text.
- Observe the Link button in the block toolbar has a 'pressed' state and styling (and an `aria-pressed="true"` attribute)
- Add an Image. Add a link to the image. Save.
- Select the image.
- Observe the Link button in the block toolbar dos not have a 'pressed' state and styling (and no `aria-pressed="true"` attribute)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by richtabor** [comment]
Related to #50893. If image and media-text use LinkControl, this would be resolved (as well as further consolidate the linking experience throughout the editor).

**Comment by afercia** [comment]
While working on this I noticed there's a  CSS problem with the `is-pressed` state. A few of the buttons used in the block toolbar have a `components-toolbar__control` class. This CSS rule:

https://github.com/WordPress/gutenberg/blob/4609399bc1c98e8311d893fcf3284c0ceec8d591/packages/components/src/toolbar/toolbar-button/style.scss#L20-L23

Introduces a problem and an inconsistent behavior. It hides the button `::before` CSS pseudo element while the button is being clicked and thus:
1. The `is-pressed` dark background isn't applied any longer while clicikng, which makes the button icon stay white on a white background so that the button looks 'empty'.
2. The focus outline disappears while clicking, which is inconsistent with all the other buttons.

Animated GIF to illustrate the second point. Using the Embed block as an example, but this can be reproduced on other buttons that use the `components-toolbar__control` class:
- The first button shows the focus outline while it is being clicked. This is the behavior of the vast majority of the buttons in the editor.
- The second button doesn't.

![focus](https://github.com/WordPress/gutenberg/assets/1682452/0592cf67-427e-43d8-8050-43ffffb142fe)

I'm going to remove this CSS rule. It was introduced [in this commit](https://github.com/WordPress/gutenberg/commit/6d946a99bb07f5a451701e9966846348eaebfbf2) as part of [the G2 redesign](https://github.com/WordPress/gutenberg/pull/19344/) and it appears to be an inconsistent remnant.

**Comment by afercia** [comment]
> Related to #50893. If image and media-text use LinkControl, this would be resolved (as well as further consolidate the linking experience throughout the editor).

Thank @richtabor it would be freat to use the same UI. In the meantime I'll push a simpke fix for the current implementation.

**Comment by afercia** [comment]
> jordesign added [Type] Enhancement and removed [Type] Bug labels 17 hours ago

In its current state, this implementation is not accessible and also the visual behavior is broken, so I'd argue this is actually a bug, not an enhancement.
